docs/architecture/adr-001-redesign-knuu-async-approach.md

```diff
@@ -0,0 +1,49 @@
+# ADR 001: Redesigning Knuu for Asynchronous Testing Approach
+
+## Status
+
+Draft
+
+## Context
+
+The existing design of `Knuu` requires exposing the addresses and ports of the target instance running alongside BitTwister, complicating the configuration process. This setup requires Knuu to orchestrate the instances, configure them, and execute the test, leading to complexities in managing traffic shaping. Moreover, as the current model relies on port forwarding, scalability issues have surfaced, causing inconsistent errors and hindering seamless test execution.
+
+## Problem
+
+The current synchronous model necessitates exposing the target instance's address and ports to BitTwister via a sidecar. This centralized approach introduces dependencies and complexities in configuring the test environment. Additionally, the reliance on a proxy approach or port forwarding poses scalability issues, resulting in inconsistent errors during test execution.
+
+## Proposed Solution
+
+### A Distributed Approach with a Playbook Model
+
+Embrace a distributed playbook model similar to Testground, allowing asynchronous orchestration and execution of test scenarios:
+
+1. **Test Scenario Definition:** Users define test scenarios in a descriptive language (e.g., Golang).
+2. **Knuu Instance Initiation:** Knuu initiates instances and assigns specific playbooks to each instance.
+3. **BitTwister Sidecar with Playbook Understanding:** BitTwister runs alongside instances with added functionality to interpret and execute the playbook's instructions.
```

Comment:
As @smuu stated in one of our syncs, we can have a scenario, where we have 1 k8s pod containing 3-4-5-6-N containers per pod. Which means we can technically have 200k~ containers while using 100 pods. 

Meaning that each of 100 pod can contain 1 BitTwister sidecar and 1800 containers, each test container will have 1 network condition behaviour. 

Thus, we can have 100k requirement achieved internally for node team